package/linux/rpm-script/postinst.sh.in

```diff
@@ -33,17 +33,26 @@ mkdir -p /var/lib/rstudio-server/proxy
 # suspend all sessions
 rstudio-server force-suspend-all
 
-# alias libcrypto.so.0.9.8 to libcrypto.so.6 for compatiblity with
-# rpms built on redhat 5
+# create openssl softlinks based off SuSE version 
 if test -f /etc/SuSE-release
 then
-  if test -d /usr/lib64
+  if test -d /lib64
   then
-     ln -s /usr/lib64/libssl.so.0.9.8 ${CMAKE_INSTALL_PREFIX}/bin/libssl.so.6
-     ln -s /usr/lib64/libcrypto.so.0.9.8 ${CMAKE_INSTALL_PREFIX}/bin/libcrypto.so.6
+    architecture_dir='/lib64'
   else
-     ln -s -f /usr/lib/libssl.so.0.9.8 ${CMAKE_INSTALL_PREFIX}/bin/libssl.so.6
-     ln -s -f /usr/lib/libcrypto.so.0.9.8 ${CMAKE_INSTALL_PREFIX}/bin/libcrypto.so.6
+    architecture_dir='/usr/lib'
+  fi
+
+  function compare_versions() { test "$(echo "$@" | tr " " "\n" | sort -V | head -n 1)" != "$1"; }
```

Comment:
For portability, you can omit the `function` keyword -- both older shells and `bash` will recgonize e.g.

    compare_version() { ... }

as a function. Not sure if this matters in practice but it looks like this script declares itself as runnable with `/bin/sh` so it's good to keep in mind.